src/IO.ts

```diff
@@ -1,6 +1,74 @@
 /**
- * @file `IO<A>` represents a synchronous computation that yields a value of type `A` and **never fails**.
- * If you want to represent a synchronous computation that may fail, please see `IOEither`.
+ * @file `IO<A>` represents a synchronous computation that yields a value of type `A` and **never fails**.  If you want
+ * to represent a synchronous computation that may fail, please see `IOEither`.  IO actions are terminated by calling
+ * their `run()` method that executes the computation and returns the result.
+ *
+ * ```ts
+ * import { IO, io } from 'fp-ts/lib/IO'
+ *
+ * const constant: IO<number> = io.of(123);
+ * constant.run()  // returns 123
+ *
+ * const random: IO<number> = new IO(() => Math.random())
+ * random.run()  // returns a random number
+ * random.run()  // returns another random number
+ * ```
+ *
+ * Below is a minimalistic example where `IO` datatype is used to implement a safe version of `console.log`.  Using
+ * `console.log` directly would break the code, resulting in both logging actions being executed when the value is not
+ * null.
+ *
+ * ```ts
+ * import { IO } from 'fp-ts/lib/IO'
+ * import { fromNullable } from 'fp-ts/lib/Option'
+ *
+ * const safeLog = (...args): IO<void> => new IO(() => console.log(...args));
+ *
+ * const logger = (input: number|null): IO<void> => fromNullable(input).fold(
+ *   safeLog('Received null'),
+ *   (value) => safeLog(`Received ${value}`),
+ * );
+ *
+ * logger(123).run();
+ * ```
+ *
+ * It is often convenient to combine several `IO` actions into one for sequential execution.
+ * You can use `sequence_(io, array)` to do this.
+ * ```ts
+ * import { IO, io } from 'fp-ts/lib/IO'
+ * import { array } from 'fp-ts/lib/Array'
+ * import { sequence_ } from 'fp-ts/lib/Foldable2v'
+ *
+ * const safeLog = (...args): IO<void> => new IO(() => console.log(...args));
+ *
+ * const logFoo: IO<void> = safeLog('foo');
+ * const logBar: IO<void> = safeLog('bar');
+ *
+ * const logFooBar: IO<void> = sequence_(io, array)([ logFoo, logBar ])
+ *
+ * logFooBar.run();
+ * ```
+ *
+ * Alternatively you could use `sequenceS(io)` to combine your computations if you are interested in results but do not
+ * care about execution order.
+ * ```ts
+ * import { IO, io } from 'fp-ts/lib/IO'
+ * import { sequenceS } from 'fp-ts/lib/Apply'
+ *
+ * interface Result {
+ *   foo: number,
+ *   bar: string,
+ * }
+ *
+ * const computations: { [K in keyof Result]: IO<Result[K]>} = {
+ *   foo: io.of(123),
+ *   bar: io.of('example')
+ * }
+ *
+ * const computation = sequenceS(io)(computations)
```

Comment:
At the moment the type `Result` is not really used, I would either add it here to show what's happening:

```ts
const computation: IO<Result> = sequenceS(io)(computations)
```

Or remove the types completely (they can be inferred) because explaining how sequence works might be better done in the `Apply` docs.